src/session_pool/session_pool.js

```diff
@@ -228,15 +261,17 @@ export class SessionPool extends EventEmitter {
     }
 
     /**
-     * Removes `Session` instance from `SessionPool`.
-     * @param {Session} session  - Session to be removed
+     * Removes retired `Session` instances from `SessionPool`.
      * @private
      */
-    _removeSession(session) {
-        const sessionIndex = this.sessions.findIndex((storedSession) => storedSession.id === session.id);
+    _removeRetiredSessions() {
+        this.sessions = this.sessions.filter((storedSession) => {
+            const sessionIsUsable = storedSession.isUsable();
```

Comment:
Probably overkill to have a separate property here. `sessionIsUsable` reads exactly the same as `session.isUsable()`

Reply:
oh yeah, I don't know what possessed me here =D Fixed